### ISSUE TITLE
FileEdit Docs Update: Changed is called when reset occurs

### ISSUE THREAD

**Issue by StephenOTT** [Megabit/Blazorise#2431]
Happy to do the PR, just need some direction:

then `Changed` event is called when the AutoReset kicks in.  As a result this means that in single file (multiple=false) FileEdit Usage, you cannot assume IFIleEntry will always contain a file (it will contain a file on first loop, and then zero files on AutoReset).

It would be good to mention this in the docs as the doc examples current use a foreach loop in the OnChanged method, but does not explain why this is the case.

The most close docs on this would be current snippet that shows:

```
...
        try
        {
            var file = e.Files.FirstOrDefault();
            if (file == null)
            {
                return;
            }
...
```

But again does not explain why the if statement is used.

Thanks

**Comment by stsrki** [comment]
The reason was that it does change when it is reset. So it makes sense to call a `Changed` event. I see a lot of people have trouble with this. Maybe it is not a bad idea to document it better and explain the reasoning.

**Comment by StephenOTT** [comment]
Agree that it makes sense. But feels like it is not obvious from default expected behaviour.

**Comment by stsrki** [comment]
@David-Moreira what do you think of this?

Should we not call the `Changed` event second time once the upload is finished? Or do we just explain it better in the documentation?

**Comment by David-Moreira** [comment]
Oh,
I have used the FileEdit before also. To me it was also intriguing the `Changed` call with no files, when the upload completes.

I think that has more to do with the expected behaviour, the user expects the `Changed ` to be called when the end-user, chooses a file, and I don't think people in general expect it to automatically clear the files after it has uploaded, therefore calling the `Changed `again. In my personal opinion the default behaviour should be AutoReset="False". That makes it so, if the user does change the behaviour, he will more quickly make a connection, between having set the AutoReset to true, and the `Changed` being called again as a Result.

As for the `Changed`, Makes sense to call `Changed` of course. The component's file state has changed, just like it says in the `Changed `description.
![image](https://user-images.githubusercontent.com/22283161/122584235-0386e580-d052-11eb-8950-a287e396e5c4.png)

Also, on a side note the text could read: 
Occurs every time the files have been changed. OR Occurs every time the file or files have been changed.
"file(s) has" sounds weird, since "files has" is not grammatically correct >.<
